website/docs/docs/collaborate/govern/model-contracts.md

```diff
@@ -86,6 +86,91 @@ When building a model with a defined contract, dbt will do two things differentl
 
+## Platform constraint support
+
+Select the adapter-specific tab for more info on adapter-specific constraint support across platforms. Constraints fall into three categories based on support and platform enforcement:
+
+- **Supported and enforced** &mdash; The model won't build if it violates the constraint.
+- **Supported and not enforced** &mdash; The platform supports specifying the type of constraint, but a model can still build even if it violates it. It's a constraint for metadata purposes only. This is common for modern cloud data warehouses and less common for legacy databases.
+- **Not supported and not enforced** &mdash; You can't specify the type of constraint for the platform.
+
+
+
+<Tabs>
+
+<TabItem value="Redshift" label="Redshift">
+
+| Constraint type | Support       | Platform enforcement |
+|:----------------|:-------------|:------------------|
+| not_null        | ✅  Supported | ✅ Enforced     |
+| primary_key     | ✅  Supported | ❌ Not enforced  |
+| foreign_key     | ✅  Supported | ❌ Not enforced  |
+| unique          | ✅  Supported | ❌ Not enforced  |
+| check           | ❌ Not supported | ❌  Not enforced |
+
+</TabItem>
+<TabItem value="Snowflake" label="Snowflake">
+
+| Constraint type | Support      | Platform enforcement |
+|:----------------|:-------------|:---------------------|
+| not_null        | ✅  Supported | ✅ Enforced     |
+| primary_key     | ✅  Supported | ❌ Not enforced  |
+| foreign_key     | ✅  Supported | ❌ Not enforced  |
+| unique          | ✅  Supported | ❌ Not enforced  |
+| check           | ❌ Not supported | ❌ Not enforced |
+
+</TabItem>
+<TabItem value="BigQuery" label="BigQuery">
+
+| Constraint type | Support       | Platform enforcement |
+|:-----------------|:-------------|:---------------------|
+| not_null        | ✅ Supported  | ✅ Enforced     |
+| primary_key     | ✅ Supported  | ✅ Enforced     |
+| foreign_key     | ✅ Supported  | ✅ Enforced     |
+| unique          | ❌ Not supported | ❌ Not enforced |
+| check           | ❌ Not supported | ❌ Not enforced |
+
+</TabItem>
+<TabItem value="Postgres" label="Postgres">
+
+| Constraint type | Support      | Platform enforcement |
+|:----------------|:-------------|:--------------------|
+| not_null        | ✅  Supported |	✅  Enforced |
+| primary_key     | ✅  Supported |	✅  Enforced |
+| foreign_key     | ✅  Supported |	✅  Enforced |
+| unique          | ✅  Supported |	✅  Enforced |
+| check           | ✅  Supported |	✅  Enforced |
+
+</TabItem>
+<TabItem value="Spark" label="Spark">
+
+`not_null` and `check` constraints are supported and enforced. Currently,  constraints are enforced only after a model builds. Because of this platform limitation, the constraints are considered `supported` but `not enforced` and not part of the "model contract" since they can't be enforced at build time. This table will change as the features evolve.
```

Comment:
Suggest simplifying a little.
```suggestion
Currently, `not_null` and `check` constraints are supported and enforced only after a model builds. Because of this platform limitation, dbt considers these constraints `supported` but `not enforced`, which means they're not part of the "model contract" since these constraints can't be enforced at build time. This table will change as the features evolve.
```